src/main/java/kohn/maze/Maze.java

```diff
@@ -1,368 +1,179 @@
+        this.rows = rows;
+        this.cols = cols;
+        maze = new Cell[rows][cols];
     }
 
-    public String[][] generateMaze(int SZ) {
-        int padding = (SZ * 2) + 1;
-        maze = new String[padding][padding];
-
-        for (int row = 0; row < padding; row++) {
-            for (int col = 0; col < padding; col++) {
-                if (col == 1 && row == 0) {
-                    maze[row][col] = "S";
-                }
-                else if (col == padding && row == 2 * SZ) {
-                    maze[row][col] = "E";
-                }
-                else if (col % 2 == 0) {
-                    if (row % 2 == 0) {
-                        maze[row][col] = "+";
-                    }
-                    else {
-                        maze[row][col] = "|";
-                    }
-                }
-                else {
-                    if (row % 2 == 0) {
-                        maze[row][col] = "-";
-                    }
-                    else {
-                        maze[row][col] = "0";
-                    }
-                }
+    private void generateMaze() {
+        int x, y;
+        for (x = 0; x < rows; x++) {
+            for (y = 0; y < cols; y++) {
+                Cell cell = new Cell(x, y);
+                maze[x][y] = cell;
+                cell.visited = false;
             }
         }
-        return maze;
-    }
 
-    public void createPath(String[][] path) {
-        Stack<Cell> stack = new Stack<>();
-        int size = (path.length - 1) / 2;
-        int allCells = size * size;
-        int visited = 1;
-        Cell current = new Cell(0, 0);
-
-        while (visited < allCells) {
-            ArrayList<String> direction = new ArrayList<>();
-            Collections.addAll(direction, "N", "E", "S", "W");
-            Collections.shuffle(direction);
-
-            String random = checkNeighbors(path, current, direction);
-
-            if (random.equals("REVERSE")) {
-                current = stack.pop();
-                continue;
+        for (x = 0; x < rows; x++) {
+            for (y = 0; y < cols; y++) {
+                assignNeighbors(maze[x][y]);
             }
-            current = removeWall(path, current, random);
-            visited = visited + 1;
-            stack.push(current);
         }
     }
 
-    private String checkNeighbors(String[][] theMaze, Cell current, ArrayList<String> directions) {
-        int size = (theMaze.length - 1) / 2;
-        int row = 2 * current.getRow() + 1;
-        int col = 2 * current.getCol() + 1;
-
-        if (directions.size() == 0) {
-            return "REVERSE";
-        }
-        String check = directions.remove(0);
-
-        if (check.equals("N")) {
-            if (current.getCol() - 1 < 0) {
-
-                return checkNeighbors(theMaze, current, directions);
+    private void createPath() {
+        int x = 0;
+        int y = 0;
+        Cell cell = maze[x][y];
+
+        while (!containsUnvisitedCells()) {
```

Comment:
`!containsUnvisitedCells()` is wrong. It's also a double negative.
It should be while there are unvisited cells.